src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -154,9 +162,14 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
                     throw new InvalidOperationException(Strings.FormatMissingParameters(string.Join(", ", unboundParams)));
                 }
 
-                rc = NativeMethods.sqlite3_step(stmt);
                 try
                 {
+                    rc = NativeMethods.sqlite3_step_blocking(Connection.DbHandle, stmt, CommandTimeout*1000);
```

Comment:
True. We could just call the sqlite3_busy_timeout as soon as the connection is opened and leave it that way for all native calls.